src/framed_write.rs

```diff
@@ -75,6 +75,8 @@ pub struct FramedWrite2<T> {
     buffer: BytesMut,
 }
 
+const SEND_HIGH_WATER_MARK: usize = 8 * 1024;
+
```

Comment:
I would be a bit wary of arbitrary numbers. Sometimes it's important to be able to send bigger batches because of performance. I have implemented AsyncRead/AsyncWrite on top of WebSockets, and there it's undesirable to flush every 8k because it will create individual 8k WebSocket Messages. Maybe this can be worked around somehow with BufWriter, but I think this needs tests to show how it behaves exactly. Also it might be good to make several pull requests allowing things that are sure to fix bugs to be merged already, and play around with uncertain changes for a bit longer.